frontend/views/Login.tsx

```diff
@@ -0,0 +1,159 @@
+          { cancelable: true }
+        );
+      });
+  };
+
+  const handleRegistration = async () => {
+    const token =
+      route.params !== undefined && route.params.token !== undefined
+        ? route.params.token
+        : "";
+
+    if (token) {
+      Admin.create(name, email, password, passwordConfirmation, token)
+        .then((response: any) => {
+          response.user = JSON.parse(response.user);
+          updateUser(response);
+          navigation.navigate("Home");
+        })
+        .catch((error) => {
+          console.error("Unsuccessful Registration", error);
+          Alert.alert(
+            "Unsuccessful Registration",
+            "helpful error message",
+            [{ text: "OK", onPress: () => console.log("OK Pressed") }],
+            { cancelable: true }
+          );
+        });
+    }
+  };
+
+  return (
+    <View
+      style={{
+        height: "100%",
+        flex: 1,
+        alignItems: "center",
+        justifyContent: "center",
+      }}
+    >
+      <LinearGradient
+        colors={["#454545", "#f0f4ef"]}
+        style={{
+          position: "absolute",
+          left: 0,
+          right: 0,
+          top: 0,
+          height: "100%",
+        }}
+      >
+        <Card style={styles.card}>
+          {route.name == "Register" ? (
+            <TextInput
+              style={styles.container}
+              mode="outlined"
+              label="name"
+              value={name}
+              onChangeText={(text) => setName(text)}
+            />
+          ) : null}
+          <TextInput
+            style={styles.container}
+            mode="outlined"
+            label="email"
+            value={email}
+            onChangeText={(text) => setEmail(text)}
+          />
+          <TextInput
+            style={styles.container}
+            mode="outlined"
+            label="password"
+            secureTextEntry={true}
+            value={password}
+            onChangeText={(text) => setPassword(text)}
+          />
+          {route.name == "Register" ? (
+            <TextInput
+              style={styles.container}
+              mode="outlined"
+              label="password confirmation"
+              secureTextEntry={true}
+              value={passwordConfirmation}
+              onChangeText={(text) => setPasswordConfirmation(text)}
+            />
+          ) : null}
+          <Button
+            icon="login"
+            mode="contained"
+            onPress={() =>
+              route.name === "Login" ? handleLogin() : handleRegistration()
+            }
+          >
+            submit
+          </Button>
+        </Card>
+      </LinearGradient>
+    </View>
+  );
+}
+
+const mapStateToProps = (state) => {
+  const { user } = state;
+  return { user };
+};
+
+const mapDispatchToProps = (dispatch) =>
+  bindActionCreators(
+    {
+      updateUser,
+    },
+    dispatch
+  );
+
+export default connect(mapStateToProps, mapDispatchToProps)(Login);
```

Comment:
This is just me not understanding Redux, can you link me some docs on what this is and why it's necessary?

Reply:
This [article](https://medium.com/@findforhad/redux-mapstatetoprops-and-mapdispatchtoprops-explained-ea48d41708dc) sums it up pretty well

Reply:
Thanks :heart:

Reply:
np thank you for all of the constructive feedback. We don't use typescript or much type checking where I work, as you can tell lol